decidim-elections/app/commands/decidim/votings/census/admin/launch_access_codes_export.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Votings
+    module Census
+      module Admin
+        # A command to launch the access codes export
+        class LaunchAccessCodesExport < Rectify::Command
+          def initialize(dataset, user)
+            @dataset = dataset
+            @user = user
+          end
+
+          # Executes the command. Broadcast this events:
+          # - :ok when everything is valid
+          # - :invalid when the user is not present
+          #
+          # Returns nothing.
+          def call
+            return broadcast(:invalid) unless valid?
+
+            ExportAccessCodesJob.perform_later(dataset, user)
+
+            UpdateDataset.call(dataset, { status: :freeze }, user)
```

Comment:
Shouldn't this be done in the `ExportAccessCodesJob`? If that job fails for some reason we won't be able to export the access codes. If you want to prevent multiple jobs, maybe add a :exporting status?

Reply:
I'm a bit hesitant about this one.
I think that we'll still have the same problem if we move this in the job and add an `exporting_codes` status to give the user feedback. If something goes wrong in the job, the status will be stuck on `exporting_codes`.

We could move everything to the job (change status to `exporting_codes` -> export codes -> change status to `freeze`) and incapsulate it in a transaction, but since the job is async it could take some time until the status changes to `exporting_codes`.

What do you think?

Reply:
Implemented the transaction solution here: https://github.com/decidim/decidim/pull/7756/commits/c5bf279e8c5d5dda291baf3ed742735e7f216263
Let me know what you think 👍🏽